service/controller/v25/resource/tccp/update.go

```diff
@@ -287,21 +292,30 @@ func (r *Resource) shouldScale(ctx context.Context, cr v1alpha1.AWSConfig, curre
 //     The size of a docker volume for worker nodes changes.
 //     The version bundle version changes (indicates updates).
 //
-func shouldUpdate(currentState, desiredState StackState) bool {
-	if currentState.MasterInstanceType != desiredState.MasterInstanceType {
-		return true
+func (r *Resource) shouldUpdate(ctx context.Context, cr v1alpha1.AWSConfig) (bool, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return false, microerror.Mask(err)
 	}
-	if currentState.WorkerDockerVolumeSizeGB != desiredState.WorkerDockerVolumeSizeGB {
-		return true
+
+	if cc.Status.TenantCluster.MasterInstance.Type != key.MasterInstanceType(cr) {
```

Comment:
The detection is here now independent from the resource state. I plan to move the scaling and updating detection to a separate package because we are going to have multiple resources managing these things and the detection has to be commonly available.